apps/common/models/i-casework-getcase.js

```diff
@@ -29,9 +30,11 @@ module.exports = class DocumentModel extends Model {
   }
 
   fetch() {
-    const options = this.requestConfig({});
-    options.qs = this.prepare();
-    options.method = 'GET';
-    return this.request(options);
+    return this.prepare().then(qs => {
```

Comment:
...and thus we assume calling this.prepare, because of super.prepare possibly being async is handled thus here.